internal/engine/watchmanager.go

```diff
@@ -133,17 +135,19 @@ func (w *WatchManager) diff(ctx context.Context, st store.RStore) (setup []Watch
 		targetsToProcess[ConfigsTargetID] = &configsTarget{dependencies: append([]string(nil), state.ConfigFiles...)}
 	}
 
+	tiltIgnoreChanged := w.tiltIgnoreContents != state.TiltIgnoreContents
+
 	for name, mnc := range w.targetWatches {
 		m, ok := targetsToProcess[name]
 		if !ok {
 			teardown = append(teardown, name)
 			continue
 		}
 
-		if !watchRulesMatch(m, mnc.target) {
+		if tiltIgnoreChanged || !watchRulesMatch(m, mnc.target) {
 			teardown = append(teardown, name)
 			setup = append(setup, m)
-			break
+			continue
```

Comment:
Isn't this continue unnecessary as the last possible statement that will be executed in this for loop?